internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1191,6 +1332,37 @@ func getObservabilityConfig(propertyName string, d *schema.ResourceData) (*vault
 	return getValidObservabilityConfig(config)
 }
 
+// if http observability information is provided, this function ensures that authentication fields are valid and returns the authentication method used
+func validateHTTPAuth(httpBasicUser string, httpBasicPassword string, httpBearerToken string) (diag.Diagnostics, *vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth, *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth) {
+	var httpConfigError diag.Diagnostics
+
+	// only one of basic or bearer authentication should be submitted
+	if httpBearerToken != "" && httpBasicUser != "" || httpBearerToken != "" && httpBasicPassword != "" {
+		httpConfigError = diag.Errorf("http configuration is invalid: either the basic or bearer authentication method can be submitted, but not both")
+	} else if httpBasicUser != "" && httpBasicPassword == "" || httpBasicUser == "" && httpBasicPassword != "" {
+		// http basic requires both the username and password to be filled
+		httpConfigError = diag.Errorf("http configuration is invalid: basic authentication requires username and password")
+	}
+
+	if httpConfigError != nil {
+		return httpConfigError, nil, nil
+	}
+
+	if httpBearerToken != "" {
+		httpBearerAuth := &vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth{
+			Token: httpBearerToken,
+		}
+		return nil, httpBearerAuth, nil
+	}
+
+	httpBasicAuth := &vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth{
```

Comment:
aren't we setting httpBasic now even if both User and Password are empty?

Reply:
Would User and Password not be empty here because httpConfigError is nil and httpBearerToken is empty if the code goes down this path?

Reply:
I believe this code should handle the case where neither basic nor bearer auth is configured, in which case the HTTP sink should be configured without any authentication settings whatsoever. However with the current implementation you would be setting basic auth with empty user and empty password if those aren't provided by the customer. In other words, leaving `httpBasicAuth==nil` will cause the Vector requests to contain a `Authentication: Basic Ogo=` when the intent was for there to be no authentication headers at all.

Or are you purposefully trying to ensure either Basic or Bearer auth is always configured? That doesn't make much sense to me

Reply:
That was my mistake. Your comment makes sense to me now.